### PR TITLE
Open source project to add message queue direction -RobustMQ

### PR DESCRIPTION
**Pull request by socutes**
@palfrey PTAL, Thanks. It looks like it meets the project inclusion criteria. There are already 51 stars, and we will continue to work hard and hope that more people can see our work.
https://github.com/robustmq/robustmq

dyz1990/sevenz-rust library no longer exists and was removed to resolve a build failure

#1782

### CODE REVIEW

**Comment by socutes** [comment]
@palfrey  Workflow needs to be approved, please have a look, thank you.

**Comment by socutes** [comment]
@palfrey Please have a look at the build error log, it seems that is deleted https://github.com/dyz1990/sevenz-rust, this project should be how to deal with this situation.

**Comment by socutes** [comment]
dyz1990/sevenz-rust library no longer exists and was removed to resolve a build failure

**Comment by socutes** [comment]
@palfrey  `Error: ErrorMessage { msg: "Not high enough metrics (None stars < 50, and None cargo downloads < 2000): 7z" }`

Why do you get this error? It looks like we met the criteria.

image

File: README.md [hunk 1]
```diff
@@ -1062,7 +1067,6 @@ See also [About Rust’s Machine Learning Community](https://medium.com/@autumn_
 ### Compression
 
 * [7z](https://7-zip.org/7z.html)
-  * [dyz1990/sevenz-rust](https://github.com/dyz1990/sevenz-rust) [[sevenz-rust](https://crates.io/crates/sevenz-rust)] - A 7z decompressor/compressor written in pure rust. [![Rust](https://github.com/dyz1990/sevenz-rust/workflows/Rust/badge.svg?branch=main)](https://github.com/dyz1990/sevenz-rust/actions)
```
Comment (palfrey) [inline README.md:1065]: This has been fixed (keeping the crate link, dropping the repo) in main, please merge that.

**Comment by palfrey** [comment]
> @palfrey `Error: ErrorMessage { msg: "Not high enough metrics (None stars < 50, and None cargo downloads < 2000): 7z" }`
> 
> Why do you get this error? It looks like we met the criteria.
> 
> image

Note the lines above
```
[2024-09-14T00:03:30Z WARN  awesome_rust] No valid github link
[2024-09-14T00:03:30Z WARN  awesome_rust] No valid crates link
```
It's not picking up the Github link for some reason, hence the wrong count. Also note, stars as None, not 0 :)

It looks like it should match the regex, and there's the `Downloading GitHub stars` line before that, so I think it's picking it up. Very weird. Not quite sure TBH, will have a look at some point.

**Comment by palfrey** [comment]
Solved it :) The "no valid github/crates link" were due to the 7z change, not robustmq. main branch now has better debug for that.

**Comment by socutes** [comment]
> 解决了:) “没有有效的 github/crates 链接”是由于 7z 变化造成的，而不是 robustmq。主分支现在有更好的调试。

**Comment by socutes** [comment]
> Solved it :) The "no valid github/crates link" were due to the 7z change, not robustmq. main branch now has better debug for that.

Thanks for your time, I'm done dealing with code conflicts.

**Pull request merged**